src/compiler/factory/nodeFactory.ts

```diff
@@ -7449,13 +7446,13 @@ function mergeEmitNode(sourceEmitNode: EmitNode, destEmitNode: EmitNode | undefi
     // `leadingComments` are concatenated with any existing leading comments on the destination
     if (leadingComments) {
         // We use `.slice()` in case `destEmitNode.leadingComments` is pushed to later
-        destEmitNode.leadingComments = addRange(leadingComments.slice(), destEmitNode.leadingComments);
+        destEmitNode.leadingComments = leadingComments.slice();

```

Comment:
I worry this results in a different inconsistency now. Before, we would attempt to merge the emit information for the old node *into* anything set on the new node prior to calling `setOriginalNode`. Now we will conditionally overwrite some properties but not others.

```ts
const oldNode = ...;
addSyntheticLeadingComment(oldNode, SyntaxKind.SingleLineCommentTrivia, "foo");

const newNode = ...;
addSyntheticLeadingComment(oldNode, SyntaxKind.SingleLineCommentTrivia, "bar");
addSyntheticTrailingComment(oldNode, SyntaxKind.SingleLineCommentTrivia, "baz");

setOriginalNode(newNode, oldNode);

// before
getSyntheticLeadingComments(newNode).length; // 2 (foo and bar)
getSyntheticTrailingComments(newNode).length; // 1 (baz)

// after
getSyntheticLeadingComments(newNode).length; // 1 (foo)
getSyntheticTrailingComments(newNode).length; // 1 (baz)
```

After this change we end up with `oldNode`'s leading comments but `newNodes` trailing comments, which seems more confusing.

Reply:
I certainly am not confident in this PR, but we do need _something_ for #59332.

Reply:
I originally considered using `appendIfUnique` to avoid duplicating but also to concatenate. Would you think that's a good fix?